src/EditorFeatures/Core/InlineRename/InlineRenameSession.cs

```diff
@@ -804,6 +804,8 @@ private async Task CommitCoreAsync(IUIThreadOperationContext operationContext, b
                 {
                     var previewService = _workspace.Services.GetService<IPreviewDialogService>();
 
+                    // The preview service needs to be called from the UI thread, since it's doing COM calls underneath.
+                    await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
```

Comment:
A more robust change here would be to make PreviewChanges async, but this is a fix we're both going to want for 17.4 Preview 2 (which is snapping shortly) and also going to consider backporting, so for _this PR_ I think this small fix is best.

Reply:
I'm happy to do a follow up in 17.4 to make it async after this change, since I agree that's the right fix.

Reply:
@ryzngard: making it async is the "following JTF" rules, although I don't know how viral that change may become, or if it'll also impact non-rename paths. Minimally if we don't we should also put a "throw if not on UI thread" check in the preview dialog service. Honestly I should have done that latter change here, but I don't want to restart CI at this point.

Reply:
Yea, that's why I'm happy with this change as is and a follow up to "do the right thing" with JTF in the future. Adding a throw could also break things that we don't know about, that technically already work by luck

Reply:
Also, if not async, it shoudl throw immediately saying it has a UI thred dependency.

Reply:
Note, the new background stacks feature found the root cause of this and we filed it here: https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1636070.

Reply:
Actually my bad, I think this might be a unique cause of the hang.